### PR TITLE
x265: enable neon from HEAD

### PR DESCRIPTION
**Pull request by kiracus**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

Apple provided a [patch](https://github.com/HandBrake/HandBrake/commit/67a490a171764eafe8eb8afb72fa1dff763a3275) to `x265`'s downstream `handshake`, `x265` [merged](https://bitbucket.org/multicoreware/x265_git/commits/4bf31dc15fb6d1f93d12ecf21fad5e695f0db5c0) it back to its `master` branch, however, it is broken as [x265_git#608](https://bitbucket.org/multicoreware/x265_git/issues/608/).

This PR borrows the patch from [x265_git#608](https://bitbucket.org/multicoreware/x265_git/issues/608/) and try to enable neon patch for Apple Silicon users.

In order to take the benefit of the neon patch, we need to do:
1. merge/apply this PR
2. `brew reinstall x265 --HEAD`, reinstall `x265` by building from the `HEAD` with the patch 
3. homebrew will trigger a rebuild of `ffmpeg`, which will used to x265 library we just rebuild. 
4. enjoy ffmpeg with the neon patch 

Ideally, when `x265` accept the patch and make a new release, we don't need to patch in `x265.rb` any more, but I make this change in advanced and hope `x265` can merge the original patch in the future

### CODE REVIEW

File: Formula/x265.rb [hunk 1]
```diff
@@ -20,6 +20,8 @@ class X265 < Formula
   depends_on "cmake" => :build
   depends_on "nasm" => :build if Hardware::CPU.intel?
 
+  patch :DATA if build.head?
```
Comment (carlocab) [inline Formula/x265.rb:23]: We don't carry patches for head builds; sorry!

Comment (kiracus) [reply Formula/x265.rb:23]: I am not sure whether I get it or not.
Currently the `brew install x265 --HEAD` is broken on Apple Silicon, and this patch will fix it. Even that, homebrew won't accept the patch?

Comment (carlocab) [reply Formula/x265.rb:23]: That is correct. We usually just remove `head` when it's broken.

**Comment by iMichka** [comment]
> Currently the brew install x265 --HEAD is broken on Apple Silicon, and this patch will fix it. Even that, homebrew won't accept the patch?

We don't really support "head" builds. They are there for testing/playing around, but in general we don't carry patches for these. We support only stable versions. 99% of our users will use the stable version.

Given this, I'll close this PR as we won't accept this patch. You can still comment here if necessary though.

**Pull request closed without merging**